JavaScriptCode.js

```diff
@@ -0,0 +1,368 @@
+    orderId++;
+    let finalarray = finalobject;
+    return finalarray;
+}
+
+let displayfun = (finalArrayObject) => {
+    let finalstring  = "";
+    finalArrayObject.map((element) => {
+        // console.log(element)
+        let finalobject = element;
+        finalstring = finalstring.concat("the Order ID : " + finalobject["orderID"] + '\n');
+        finalstring = finalstring.concat("the total number of an bags ordered : " + finalobject["totalNumber"] + '\n');
+        finalstring = finalstring.concat("the total price of coffee : " + finalobject["totalCoffePrice"] + currency + '\n');
+        let boxesarray = boxesarrayfun(finalobject["totalNumber"]);
+        let numberofboxestring = "the number of boxes used from each size and their respective prices : \n";
+        boxesarray.filter((element) => {
+             if(element.quantity_of_Size > 0){
+                numberofboxestring = numberofboxestring.concat('    the number of ' + element.size + ' box|boxes is : ' + element.quantity_of_Size + " with quantity of bags : " + element.numbers_of_Bags + " (each box/ " );
+                element.size === 'Large'? numberofboxestring = numberofboxestring.concat('can contain 20'):element.size === 'Medium'?numberofboxestring = numberofboxestring.concat('can contain 10'):numberofboxestring = numberofboxestring.concat('can contain 5');
+                return  numberofboxestring = numberofboxestring.concat(" bags), where the box|boxes price : " + element.price_of_Boxes + '' + currency+ '\n');
+             }
+             else{
+                return numberofboxestring = numberofboxestring.concat('')
+             }
+        });
+        let removelastbreakline = numberofboxestring;
+        removelastbreakline = removelastbreakline.substring(0, removelastbreakline.length - 1);
+        finalstring = finalstring.concat(removelastbreakline + '\n');
+        finalstring = finalstring.concat("the total order price Bages(" + finalobject["totalBagsPrice"] + currency + ") + coffee(" + finalobject["totalCoffePrice"] + currency + "): " + finalobject["totalOrderPrice"] + currency + '\n');
+        finalstring = finalstring.concat('\n');
+    });
+    return finalstring;
+}
+let finalArrayObject = []; finalArrayObject = [order(myordernumber)];
+console.log("///////////////////    Part A :    ///////////////////");
+console.log(displayfun(finalArrayObject));
+/////////////////////////////////////////////////////
+// End of part A
+/////////////////////////////////////////////////////
+
+
+
+/////////////////////////////////////////////////////
+// Start of part B
+/////////////////////////////////////////////////////
+
+let partBfun = (del, valuesarray) => {
+    let GivenValuefun = (valuesarray) => {
+        let arraygivenvalue = [];
+        valuesarray.map((element) => {
+            arraygivenvalue.push(order(element));
+        });
+        return arraygivenvalue;
+    };
+    let arraygivenvalue = GivenValuefun(valuesarray);
```

Comment:
for all the app, let's make sure to use const when it isn't changed, and let when we want to change

Reply:
noted

Reply:
done